core/app/components/forms/elements/ContextEditorElement.tsx

```diff
@@ -17,81 +18,73 @@ import { ContextEditorClient } from "../../ContextEditor/ContextEditorClient";
-	},
-	content: [
-		{
-			type: "paragraph",
-			attrs: {
-				id: null,
-				class: null,
-			},
-		},
-	],
-};
+const EditorFormElement = memo(
+	function EditorFormElement({
+		field,
+		label,
+		help,
+	}: {
+		field: ControllerRenderProps<FieldValues, string>;
+		label: string;
+		help?: string;
+	}) {
+		const formElementToggle = useFormElementToggleContext();
+		const { pubs, pubTypes, pubId, pubTypeId } = useContextEditorContext();
 
-const EditorFormElement = ({
-	label,
-	help,
-	onChange,
-	initialValue,
-	disabled,
-}: {
-	label: string;
-	help?: string;
-	onChange: (state: any) => void;
-	initialValue?: Node;
-	disabled?: boolean;
-}) => {
-	const { pubs, pubTypes, pubId, pubTypeId } = useContextEditorContext();
-	const [initialDoc] = useState(initialValue);
+		const f = useMemo(() => {
+			return field;
+		}, []);
 
-	if (!pubId || !pubTypeId) {
-		return null;
-	}
+		const [initialHtml] = useState(f.value);
 
-	return (
-		<FormItem>
-			<FormLabel className="flex">{label}</FormLabel>
-			<div className="w-full">
-				<FormControl>
-					<ContextEditorClient
-						pubId={pubId}
-						pubs={pubs}
-						pubTypes={pubTypes}
-						pubTypeId={pubTypeId}
-						onChange={(state) => {
-							// Control changing the state more granularly or else the dirty field will trigger on load
-							// Since we can't control the dirty state directly, even this workaround does not handle the case of
-							// if someone changes the doc but then reverts it--that will still count as dirty since react-hook-form is tracking that
-							const hasChanged = docHasChanged(initialDoc ?? EMPTY_DOC, state);
-							if (hasChanged) {
-								onChange(state);
-							}
-						}}
-						initialDoc={initialDoc}
-						disabled={disabled}
-						className="max-h-96 overflow-scroll"
-					/>
-				</FormControl>
-			</div>
-			<FormDescription>{help}</FormDescription>
-			<FormMessage />
-		</FormItem>
-	);
-};
+		if (!pubId || !pubTypeId) {
+			return <></>;
+		}
+		const disabled = !formElementToggle.isEnabled(f.name);
+
+		return (
+			<FormItem>
+				<FormLabel className="flex">{label}</FormLabel>
+				<div className="w-full">
+					<FormControl>
+						<ContextEditorClient
+							pubId={pubId}
+							pubs={pubs}
+							pubTypes={pubTypes}
+							pubTypeId={pubTypeId}
+							onChange={(state, initialDoc, initialHtml) => {
+								// Control changing the state more granularly or else the dirty field will trigger on load
+								// Since we can't control the dirty state directly, even this workaround does not handle the case of
+								// if someone changes the doc but then reverts it--that will still count as dirty since react-hook-form is tracking that
+								const hasChanged = docHasChanged(initialDoc ?? EMPTY_DOC, state);
+								if (hasChanged) {
+									f.onChange(serializeProseMirrorDoc(state.doc));
+								}
+							}}
+							initialHtml={initialHtml}
+							disabled={disabled}
+							className="max-h-96 overflow-scroll"
+						/>
```

Comment:
so this is a bit cringe: we are still intermittently setting the value of a `RichText` field to the prosemirror tree in the form. i thought it a bit too much to convert the prosemirror tree to `html` on each change.

what's annoying is that this sort of breaks the nice dichotomy i had where only the actual `ContextEditor` knew that the `RichText` field should be converted to a `ProseMirror` tree, everything else just thinks it's HTML. this idea is maybe flawed in general.